frontend/markdown-pdf/build.js

```diff
@@ -113,77 +157,668 @@ Promise.all([
-                        },
-                        displayHeaderFooter: true,
-                        headerTemplate: `
-                            <style>${css}</style>
-                            <header>
-                                <img src="data:image/svg+xml;base64,${logoSVG.toString(
-                                    'base64'
-                                )}" class="logo" />
-                                ${filePath.replace(/\.md$/, '')}
-                            </header>`,
-                        footerTemplate: `
-                            <style>${css}</style>
-                            <footer>
-                                <span class="pageNumber"></span> / <span class="totalPages"></span>
-                            </footer>`,
-                    });
-                }),
-            ]).then(([, pdf]) =>
-                fs.writeFile(
-                    path.join(outputDir, filePath.replace('.md', '.pdf')),
-                    pdf
-                )
+                    pdfs.map(pdf =>
+                        fs.writeFile(
+                            path.join(
+                                outputDir,
+                                filePath.replace('.md', '.pdf')
+                            ),
+                            pdf
+                        )
+                    ),
+                    renderEmptyPage(browser, 'A6'), // Generate an empty page we can use in our booklets
+                ])
             )
         )
-    ).finally(() => browser.close())
+    )
+        .then(([[, emptyPage]]) =>
+            Promise.all(
+                // Loop over all distinct markdown files
+                Array.from(languages.keys(), async manual => {
+                    // Check whether the markdown file exists in the root folder
+                    if (fs.existsSync(path.join(inputDir, manual))) {
+                        // Parse Frontmatter for config extraction
+                        const frontmatter = vuepressUtil.parseFrontmatter(
+                            await fs.readFile(
+                                path.join(inputDir, manual),
+                                'utf8'
+                            )
+                        );
+
+                        // Combine all config elements in the frontmatter into 1 dict
+                        var config = {};
+                        if (frontmatter.data.config) {
+                            frontmatter.data.config.forEach(elem => {
+                                config[elem.name] = elem.content;
+                            });
+                        } else {
+                            // If there is no config, this md file does not need to be converted to a booklet
+                            return;
+                        }
+
+                        // If we don't need to generate a booklet for this md file, skip this file
+                        if (!config.generate_booklet) {
```

Comment:
Is er een use case voor het uitzetten van booklet generation terwijl de user wel een booklet config specificeert? Anders is het waarschijnlijk netter om hier geen aparte duplicate flag voor te hebben, maar gewoon "is er een booklet config, dan maken we een booklet".

Reply:
Fair point